src/vma/iomux/epoll_wait_call.cpp

```diff
@@ -85,45 +85,45 @@ int epoll_wait_call::get_current_events()
 	i = r = w = m_n_all_ready_fds;
 	socket_fd_api *p_socket_object;
 	epoll_fd_rec fd_rec;
-	ep_ready_fd_map_t::iterator iter = m_epfd_info->m_ready_fds.begin();
+	list_iterator_t<socket_fd_api, socket_fd_api::ep_ready_fd_node_offset> iter = m_epfd_info->m_ready_fds.begin();
 	while (iter != m_epfd_info->m_ready_fds.end() && i < m_maxevents) {
-		ep_ready_fd_map_t::iterator iter_cpy = iter; // for protection needs
+		list_iterator_t<socket_fd_api, socket_fd_api::ep_ready_fd_node_offset> iter_cpy = iter; // for protection needs
 		++iter;
-		p_socket_object = fd_collection_get_sockfd(iter_cpy->first);
+		p_socket_object = iter_cpy.getPtr();
```

Comment:
In original code you get p_socket_object by: 
fd_collection_get_sockfd(iter_cpy->first);
In your code you get p_socket_object by:
iter_cpy.getPtr();

Can you think of a way of getting fd from iter_cpy, then align with the original code and get p_socket_object by: 
fd_collection_get_sockfd(fd);....